amy/dashboard/views.py

```diff
@@ -245,57 +245,135 @@ def training_progress(request):
 
 
 class GetInvolvedCreateView(LoginRequiredMixin, RedirectSupportMixin, AMYCreateView):
-    # permission_required = TODO
     model = TrainingProgress
     form_class = GetInvolvedForm
     template_name = "get_involved_form.html"
     success_url = reverse_lazy("training-progress")
     success_message = (
-        "Thank you. Your Get Involved submission will be reviewed within 7-10 days."
+        "Thank you. Your Get Involved submission will be evaluated within 7-10 days."
     )
 
+    def user_may_create_submission(self, user):
+        """The user may create a submission if either of the conditions below are met:
+        1. no progress exists for the Get Involved step for this user
+        2. all existing progress for the Get Involved step for this user has state "a"
+        """
+        get_involved = TrainingRequirement.objects.get(name="Get Involved")
+        existing_progresses = TrainingProgress.objects.filter(
+            requirement=get_involved, trainee=user
+        )
+        num_existing = existing_progresses.count()
+        if num_existing == 0:
+            return True
+        else:
+            num_asked_to_repeat = existing_progresses.filter(state="a").count()
+            if num_asked_to_repeat == num_existing:
+                return True
+            else:
+                return False
+
     def get_context_data(self, **kwargs) -> dict:
         context = super().get_context_data(**kwargs)
         context["title"] = str("Submit your Get Involved activity")
         return context
 
     def post(self, request):
         self.object = None
+
         base_training_progress = TrainingProgress(
             trainee=request.user,
             state="n",  # not evaluated yet
             requirement=TrainingRequirement.objects.get(name="Get Involved"),
         )
-        get_involved_form = GetInvolvedForm(
-            data=request.POST, instance=base_training_progress
-        )
+        # select data specifically to prevent user from submitting extra args
+        data = {
+            "involvement_type": request.POST.get("involvement_type"),
+            "date": request.POST.get("date"),
+            "url": request.POST.get("url"),
+            "trainee_notes": request.POST.get("trainee_notes"),
+        }
+        get_involved_form = GetInvolvedForm(data=data, instance=base_training_progress)
 
-        if get_involved_form.is_valid():
+        user_may_create_submission = self.user_may_create_submission(request.user)
+        if get_involved_form.is_valid() and user_may_create_submission:
             return self.form_valid(get_involved_form)
-
         else:
+            if not user_may_create_submission:
+                self.form_invalid_message = (
+                    "You already have an existing submission. "
+                    "You may not create another submission unless your previous "
+                    'submission has the status "asked to repeat."'
+                )
```

Comment:
This can also be handled by form `clean()` method:

```python
...

    def clean(self) -> None:
        if not self.user_may_create_submission():
            raise ValidationError("You already have an existing submission.")

```